wallet/src/Pos/Wallet/Web/Backup.hs

```diff
@@ -11,30 +11,46 @@ import           Universum
 
 import qualified Data.HashMap.Strict as HM
 import qualified Data.SemVer as V
-
 import           Pos.Crypto (EncryptedSecretKey)
+import           Pos.Crypto.Signing.Safe (emptyPassphrase, safeKeyGen)
+import           Pos.Wallet.Web.ClientTypes (AccountId (..), CAccountMeta (..),
+                                             CId, CWalletMeta (..), Wal,
+                                             CWalletAssurance (..))
 import           Pos.Util.Util (maybeThrow)
 import           Pos.Wallet.Web.Account (AccountMode, getSKById)
-import           Pos.Wallet.Web.ClientTypes (AccountId (..), CAccountMeta (..), CId,
-                                             CWalletMeta (..), Wal)
 import           Pos.Wallet.Web.Error (WalletError (..))
 import           Pos.Wallet.Web.State (getAccountMeta, getWalletMeta)
 import           Pos.Wallet.Web.Util (getWalletAccountIds)
+import           Test.QuickCheck (Arbitrary (..))
 
 currentBackupFormatVersion :: V.Version
 currentBackupFormatVersion = V.initial & V.major .~ 1
 
-newtype WalletMetaBackup = WalletMetaBackup CWalletMeta
-newtype AccountMetaBackup = AccountMetaBackup CAccountMeta
+newtype WalletMetaBackup = WalletMetaBackup CWalletMeta deriving Show
+newtype AccountMetaBackup = AccountMetaBackup CAccountMeta deriving Show
 
 data WalletBackup = WalletBackup
     { wbSecretKey :: !EncryptedSecretKey
     , wbMeta      :: !WalletMetaBackup
     , wbAccounts  :: !(HashMap Int AccountMetaBackup)
-    }
+    } deriving Show
 
 data TotalBackup = TotalBackup WalletBackup
 
+instance Arbitrary WalletBackup where
+    arbitrary = do
+        cwNameT <- arbitrary
+        caNameT <- arbitrary
+        wbInt   <- arbitrary
+        let hm = HM.empty
+        (_, esk) <- safeKeyGen emptyPassphrase
+        let wMetaBackup = WalletMetaBackup $ CWalletMeta { cwName = (cwNameT :: Text ), cwAssurance = CWANormal, cwUnit = 1 }
```

Comment:
you could also do `arbitrary` for assurance. By adding arbitrary for assurance:
```haskell
data CWalletAssurance = CWAStrict | CWANormal 
  deriving (Enum, Bounded)

instance Arbitrary CWalletAssurance where
  arbitrary = element [minBound..maxBound]
```

or by simply inserting in your code:
```
  cwAssurance <- element [minBound .. maxBound]
```

Also, I believe we have a syntax sugar on where you should do something like with the above:
```
instance Arbitrary WalletBackup where
+    arbitrary = do
+        cwName <- arbitrary
+        caName <- arbitrary
+        wbInt   <- arbitrary
+        let hm = HM.empty
+        (_, esk) <- safeKeyGen emptyPassphrase
+        let wMetaBackup = WalletMetaBackup $ CWalletMeta {  cwAssurance = CWANormal, cwUnit = 1, ... }
```
and `cwName` will be picked automatically.

Reply:
Hey @akegalj thanks for review!
```
cwAssurance <- element [minBound .. maxBound]
```
Did you mean `elements` ?  -> [elements](https://www.stackage.org/haddock/lts-9.14/QuickCheck-2.9.2/src/Test.QuickCheck.Gen.html#elements)
> Also, I believe we have a syntax sugar on

I am not sure what you mean, use `RecordWildCards` ?
```
instance Arbitrary WalletBackup where
    arbitrary = do
        cwNameT <- arbitrary
        caNameT <- arbitrary
        wbInt   <- arbitrary
        cwAssurance <- elements [minBound .. maxBound]
        (_, esk) <- safeKeyGen emptyPassphrase
        let cwUnit = 1
            cwName = (cwNameT :: Text )
            caName = (caNameT :: Text )
            wMetaBackup = WalletMetaBackup $ CWalletMeta {..}
        return $ WalletBackup
            { wbSecretKey = esk
            , wbMeta = wMetaBackup
            , wbAccounts = HM.singleton wbInt (AccountMetaBackup CAccountMeta {..})
            }
```

Reply:
> Did you mean elements ? -> elements

yes

> I am not sure what you mean, use RecordWildCards ?

yes

The example bellow should work:
```
Prelude> :set -XRecordWildCards
Prelude> data Test = Test { t1 :: Int, t2 :: Int } deriving Show
Prelude> t1 = 2
Prelude> Test { t2 = 2, ..}

<interactive>:15:1: warning: [-Wmissing-fields]
    • Fields of 'Test' not initialised: t1
    • In the expression: Test {t2 = 2, ..}
      In an equation for 'it': it = Test {t2 = 2, ..}

<interactive>:15:1: warning: [-Wmissing-fields]
    • Fields of 'Test' not initialised: t1
    • In the expression: Test {t2 = 2, ..}
      In an equation for 'it': it = Test {t2 = 2, ..}
Test {t1 = *** Exception: <interactive>:15:1-18: Missing field in record construction t1
```
but for some unfortunate reason it doesn't work in the repl. Probably its a repl bug. It should return `Test {t1 = 2, t2 = 2}`

Reply:
I get it, thanks! So I believe all is ready now